app/addons/documents/index-results/actions.js

```diff
@@ -71,7 +71,49 @@ function (app, FauxtonAPI, ActionTypes, Stores, HeaderStores, HeaderActions, Doc
       });
     },
 
+    newMangoResultsList: function (options) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.INDEX_RESULTS_NEW_RESULTS,
+        options: options
+      });
+    },
+
+    runMangoFindQuery: function (options) {
+      var query = JSON.parse(options.queryCode),
+          collection = indexResultsStore.getCollection();
+
+      this.clearResults();
+
+      return collection
+        .runQuery(options.queryCode)
+        .then(function () {
+          this.resultsListReset();
+          this.newMangoResultsList({
+            collection: collection,
+            isListDeletable: indexResultsStore.isListDeletable(),
+            query: options.queryCode,
+            typeOfIndex: 'mango',
+            textEmptyIndex: 'No Results Found!',
+            bulkCollection: Documents.BulkDeleteDocCollection
+          });
+        }.bind(this), function (res) {
+          FauxtonAPI.addNotification({
+            msg: res.reason,
+            clear:  true,
+            type: 'error'
+          });
+
+          this.resultsListReset();
+        }.bind(this));
+    },
+
     reloadResultsList: function () {
+      if (indexResultsStore.getTypeOfIndex() === 'mango') {
+        return this.runMangoFindQuery({
```

Comment:
i wish i could!

problem is that it is called internally from our bulkDeleteAction: https://github.com/robertkowalski/couchdb-fauxton/blob/mango-editor/app/addons/documents/index-results/actions.js#L168